common/src/main/java/org/jboss/pnc/common/json/ConfigurationJSONParser.java

```diff
@@ -0,0 +1,31 @@
+package org.jboss.pnc.common.json;
+
+import java.io.File;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+/**
+ * @author pslegr
+ *
+ */
+
+public class ConfigurationJSONParser<T extends AbstractModuleConfig> {
+
+    public T parseJSONConfig(File configFile, Class<T> classType) throws ConfigurationParseException {
+        try {
+            // read from file, convert it to config module class
+            ObjectMapper mapper = new ObjectMapper();
+            ModuleConfigJson json_config = mapper.readValue(configFile, ModuleConfigJson.class);
+
+            for (AbstractModuleConfig config : json_config.getConfigs()) {
+              if(config.getClass().isAssignableFrom(classType)) {
```

Comment:
We can also Cast without checking. If the type is incorrect we will end up in catch block and the resulting exception would be the same.